core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3608,6 +3608,9 @@ public void mount(AlluxioURI alluxioPath, AlluxioURI ufsPath, MountContext conte
         }
         mMountTable.checkUnderWritableMountPoint(alluxioPath);
 
+        if (context.getOptions().getRemount()) {
```

Comment:
check `hasRemount()` first, for backward compatibility?

Reply:
<img width="558" alt="image" src="https://github.com/Alluxio/alluxio/assets/17329931/edec04cd-1295-4ec3-adfa-279aeb38d1bc">

This is a screenshot of generated class `MountPOptions`, it return the `readOnly_` directly.